sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs

```diff
@@ -0,0 +1,114 @@
+                startTime: DateTime.UtcNow);
+            activity.Stop();
+
+            var httpUrl = "https://www.foo.bar/search";
+            activity.SetStatus(Status.Ok);
+            activity.SetTag(SemanticConventions.AttributeHttpUrl, httpUrl); // only adding test via http.url. all possible combinations are covered in HttpHelperTests.
+            activity.SetTag(SemanticConventions.AttributeHttpStatusCode, null);
+            var requestData = TelemetryPartB.GetRequestData(activity);
+
+            Assert.Equal(activity.DisplayName, requestData.Name);
+            Assert.Equal(activity.Context.SpanId.ToHexString(), requestData.Id);
+            Assert.Equal(httpUrl, requestData.Url);
+            Assert.Equal("0", requestData.ResponseCode);
+            Assert.Equal(activity.Duration.ToString("c", CultureInfo.InvariantCulture), requestData.Duration);
+            Assert.Equal(activity.GetStatus() != Status.Error, requestData.Success);
+            Assert.Null(requestData.Source);
+            Assert.True(requestData.Properties.Count == 1); //Because of otel_statuscode attribute for activity status. todo: do not add all tags to PartC
+            Assert.True(requestData.Measurements.Count == 0);
+        }
+
+        [Theory]
+        [InlineData("200")]
+        [InlineData(null)]
+        public void ValidateHttpRequestDataResponseCode(string httpStatusCode)
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                parentContext: new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded),
+                startTime: DateTime.UtcNow);
+
+            var httpResponsCode = httpStatusCode ?? "0";
+            activity.SetTag(SemanticConventions.AttributeHttpUrl, "https://www.foo.bar/search");
+            activity.SetTag(SemanticConventions.AttributeHttpStatusCode, httpStatusCode);
+            var requestData = TelemetryPartB.GetRequestData(activity);
+
+            Assert.Equal(httpResponsCode, requestData.ResponseCode);
+        }
+
+        [Theory]
+        [InlineData("Ok")]
+        [InlineData("Error")]
+        [InlineData("Unset")]
+        public void ValidateRequestDataSuccess(string activityStatus)
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                parentContext: new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded),
+                startTime: DateTime.UtcNow);
+
+            if (activityStatus == "Ok")
```

Comment:
`Status` could be used as test args. In that way we could avoid condition check here.

Reply:
it does not allow me to pass Status.Ok etc. as args. Gives this [error](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0182?f1url=%3FappId%3Droslyn%26k%3Dk(CS0182))